include/fdreadoutlibs/wib/RAWWIBTriggerPrimitiveProcessor.hpp

```diff
@@ -298,6 +160,188 @@ void tp_unpack(frame_ptr fr)
   }
 
 
+void unpack_tpframe_version_1(frame_ptr fr)
+{
+  auto& srcbuffer = fr->get_data();
+  int num_elem = fr->get_raw_tp_frame_chunksize();
+
+  if (num_elem == 0) {
+    TLOG_DEBUG(TLVL_WORK_STEPS) << "No raw WIB TP elements to read from buffer! ";
+    return;
+  }
+  if (num_elem % RAW_WIB_TP_SUBFRAME_SIZE != 0) {
+    TLOG_DEBUG(TLVL_WORK_STEPS) << "Raw WIB TP elements not multiple of subframe size (3)! ";
+    return;
+  }
+
+  int offset = 0;
+  rwtp = std::make_unique<detdataformats::wib::RawWIBTp>();
+  while (offset <= num_elem) {
+
+    if (offset == num_elem) break;
+
+    // Count number of subframes in a TP frame
+    int n = 1;
+    while (reinterpret_cast<types::TpSubframe*>(((uint8_t*)srcbuffer.data()) // NOLINT
+           + offset + (n-1)*RAW_WIB_TP_SUBFRAME_SIZE)->word3 != 0xDEADBEEF) {
+      n++;
+    }
+
+    int bsize = n * RAW_WIB_TP_SUBFRAME_SIZE;
+    std::vector<char> tmpbuffer;
+    tmpbuffer.reserve(bsize);
+    int nhits = n - 2;
+
+    // add header block 
+    ::memcpy(static_cast<void*>(tmpbuffer.data() + 0),
+             static_cast<void*>(srcbuffer.data() + offset),
+             RAW_WIB_TP_SUBFRAME_SIZE);
+
+    // add pedinfo block 
+    ::memcpy(static_cast<void*>(tmpbuffer.data() + RAW_WIB_TP_SUBFRAME_SIZE),
+             static_cast<void*>(srcbuffer.data() + offset + (n-1)*RAW_WIB_TP_SUBFRAME_SIZE),
+             RAW_WIB_TP_SUBFRAME_SIZE);
+
+    // add TP hits 
+    ::memcpy(static_cast<void*>(tmpbuffer.data() + 2*RAW_WIB_TP_SUBFRAME_SIZE),
+             static_cast<void*>(srcbuffer.data() + offset + RAW_WIB_TP_SUBFRAME_SIZE),
+             nhits*RAW_WIB_TP_SUBFRAME_SIZE);
+
+    dunedaq::detdataformats::wib::RawWIBTp* rwtps =
+           static_cast<dunedaq::detdataformats::wib::RawWIBTp*>( malloc(
+           sizeof(dunedaq::detdataformats::wib::TpHeader) + nhits * sizeof(dunedaq::detdataformats::wib::     TpData)
+           ));
+    rwtp.release( );
+    rwtp.reset( rwtps );
+    rwtp->set_nhits(nhits);
+
+    ::memcpy(static_cast<void*>(&rwtp->m_head),
+             static_cast<void*>(tmpbuffer.data() + 0),
+             2*RAW_WIB_TP_SUBFRAME_SIZE);
+
+    for (int i=0; i<nhits; i++) {
+      ::memcpy(static_cast<void*>(&rwtp->m_blocks[i]),
+               static_cast<void*>(tmpbuffer.data() + (2+i)*RAW_WIB_TP_SUBFRAME_SIZE),
+               RAW_WIB_TP_SUBFRAME_SIZE);
+    }
+
+    // old format lacks number of hits
+    rwtp->set_nhits(nhits); // explicitly set number of hits in new format
+
+    int debug_nhits = rwtp->get_nhits();
+    uint16_t debug_padding = rwtp->get_padding_3(); // NOLINT
+    uint16_t debug_wire = rwtp->m_head.m_wire_no; // NOLINT
+    TLOG_DEBUG(TLVL_WORK_STEPS) << "Number of hits, padding, wire " << debug_nhits << ", " << debug_padding   << ", " << debug_wire;
+
+    // raw recording 
+    tp_dump_frame();
+
+    // stitch TP hits
+    tp_stitch();
+    offset += (2+nhits)*RAW_WIB_TP_SUBFRAME_SIZE;
+  }
+}
+
+void unpack_tpframe_version_2(frame_ptr fr)
```

Comment:
do we need this function ?  there is no production firmware running version 2 yet.

Reply:
good point! I assume that now we can take this decision here and discard any handling of the new TP fw format, and target the next real running setup which will use the old TP format. 
 
The new TP format was used for development only. 

Would it be a compromise to leave the new TP format code but completely remove it from the execution path? Or perhaps it would be better and cleaner to delete it?

Reply:
I don't think this is a big problem, so let's leave it.